base/operators.jl

```diff
@@ -946,6 +947,40 @@ struct Fix2{F,T} <: Function
 end
 
 (f::Fix2)(y) = f.f(y, f.x)
+=#
+
+interleave(bind::Tuple{}, args::Tuple{}) = ()
+interleave(bind::Tuple{}, args::Tuple) = error("more args than positions")
+interleave(bind, args) = _interleave(first(bind), tail(bind), args)
+
+# `nothing` indicates a position to be bound
+_interleave(firstbind::Nothing, tailbind::Tuple, args::Tuple) = (
+  first(args), interleave(tailbind, tail(args))...)
+
+# allow escaping of e.g. `nothing`
+_interleave(firstbind::Some{T}, tailbind::Tuple, args::Tuple) where T = (
+  something(firstbind), interleave(tailbind, args)...)
+
+_interleave(firstbind::T, tailbind::Tuple, args::Tuple) where T = (
+  firstbind, interleave(tailbind, args)...)
+
+struct Bind{F, A}
```

Comment:
This is required for backward compatibility:

```suggestion
struct Bind{F, A} <: Function
```